gossip/src/push_active_set.rs

```diff
@@ -77,32 +181,62 @@ impl PushActiveSet {
             .iter()
             .map(|node| get_stake_bucket(stakes.get(node)))
             .collect();
-        // (k, entry) represents push active set where the stake bucket of
-        //     min stake of {this node, crds value owner}
-        // is equal to `k`. The `entry` maintains set of gossip nodes to
-        // actively push to for crds values belonging to this bucket.
-        for (k, entry) in self.0.iter_mut().enumerate() {
-            let weights: Vec<u64> = buckets
-                .iter()
-                .map(|&bucket| {
-                    // bucket <- get_stake_bucket(min stake of {
-                    //  this node, crds value owner and gossip peer
-                    // })
-                    // weight <- (bucket + 1)^2
-                    // min stake of {...} is a proxy for how much we care about
-                    // the link, and tries to mirror similar logic on the
-                    // receiving end when pruning incoming links:
-                    // https://github.com/solana-labs/solana/blob/81394cf92/gossip/src/received_cache.rs#L100-L105
-                    let bucket = bucket.min(k) as u64;
-                    bucket.saturating_add(1).saturating_pow(2)
-                })
-                .collect();
-            entry.rotate(rng, size, num_bloom_filter_items, nodes, &weights);
+
+        match self.mode {
```

Comment:
can this block be refactored to avoid replicode? seems like the map closure is the only difference